src/apps/answers/crud/answers.py

```diff
@@ -234,6 +234,13 @@ async def get_respondents_submit_dates(
         query = query.where(func.date(AnswerSchema.created_at) >= filters.from_date)
         query = query.where(func.date(AnswerSchema.created_at) <= filters.to_date)
         query = query.where(AnswerSchema.applet_id == applet_id)
+        query = query.where(
+            or_(
+                AnswerSchema.id_from_history_id(AnswerSchema.activity_history_id) == str(filters.activity_or_flow_id),
+                AnswerSchema.id_from_history_id(AnswerSchema.flow_history_id) == str(filters.activity_or_flow_id),
+            ),
+        )
+
```

Comment:
The have a similiar code in `get_target_subject_ids_by_respondent` function and change it because it was not working properly, because `activity_history_id` or `flow_history_id` can be None, we use the example I sent in other comment

Reply:
I'm going to approve it, but you should test it